### DISCUSSION TITLE
"Invalid time" for all clips in "Events"

### DISCUSSION THREAD

**Discussion by thinkloop** [blakeblackshear/frigate#5621] [Ask A Question]
Hello, all the clips in "Events" are marked with "Invalid time", any idea what that's about?

![image](https://user-images.githubusercontent.com/118814/222179744-446037be-7bde-4e50-b85f-021fc2ed32db.png)

**Comment by NickM-27** [comment]
This may be fixed in the next beta, not sure why this is happening but we've had a number of changes with this.

**Comment by thinkloop** [comment]
Hello, I installed all the betas since the original post and now rc1 but I still see "invalid time" for all events - is this a known issue?

![image](https://user-images.githubusercontent.com/118814/228302869-d90d54fa-8610-4e2e-8b8c-81826a1063b5.png)

**Reply by NickM-27** [reply]
no it is not a known issue, if you check the network calls in browser dev tools what do the start times look like? most likely this is a timezone issue perhaps

**Reply by thinkloop** [reply]
You're right there are exception in the date code:

![image](https://user-images.githubusercontent.com/118814/228306095-c5b15f8e-7318-4e94-aeb3-8258e81a71ed.png)

The ubuntu box shows this for `timedatectl`:

![image](https://user-images.githubusercontent.com/118814/228306401-c345db26-efe6-4324-b490-a96cfb29b53b.png)

Is that ok?

The json seems to have valid times:

```
{
        "area": null,
        "box": null,
        "camera": "lionel-package",
        "end_time": 1680019858.984779,
        "false_positive": null,
        "has_clip": true,
        "has_snapshot": true,
        "id": "1680019847.756642-stk9mi",
        "label": "person",
        "plus_id": null,
        "ratio": null,
        "region": null,
        "retain_indefinitely": false,
        "start_time": 1680019845.756642,
        "sub_label": null,
        "top_score": 0.7890625,
        "zones": []
    },
    {
        "area": null,
        "box": null,
        "camera": "lionel-package",
        "end_time": 1680011194.141544,
        "false_positive": null,
        "has_clip": true,
        "has_snapshot": true,
        "id": "1680011181.404098-so2a10",
        "label": "person",
        "plus_id": null,
        "ratio": null,
        "region": null,
        "retain_indefinitely": false,
        "start_time": 1680011179.404098,
        "sub_label": null,
        "top_score": 0.83984375,
        "zones": []
    },
```

**Reply by NickM-27** [reply]
Where is that first screenshot from?

**Reply by thinkloop** [reply]
It's from the Edge console when pause on exceptions is on

**Reply by thinkloop** [reply]
And you seem to be right it's a zone issue:

![image](https://user-images.githubusercontent.com/118814/228308763-9ebaebe6-a5b8-4bdc-8dd2-edb25dbbbae1.png)

**Reply by thinkloop** [reply]
The timezone is JS is Toronto, whereas the ubuntu box is New York:

![image](https://user-images.githubusercontent.com/118814/228309048-1fa489fb-b916-4039-aadf-21f73b1faa24.png)

**Reply by thinkloop** [reply]
It catches here at invalid time:

![image](https://user-images.githubusercontent.com/118814/228309258-c4ad55fe-d34b-41da-bd09-f8ac8bd5f235.png)

**Reply by thinkloop** [reply]
```
timeZone: u || Intl.DateTimeFormat().resolvedOptions().timeZone,
```

The timezone ended up being "None" because "None" is non-null and in the above code `u === 'None'`

**Reply by thinkloop** [reply]
Which comes from this line:

![image](https://user-images.githubusercontent.com/118814/228310131-6cb0dde6-a523-4df2-8deb-91c2131b4ad9.png)

**Reply by NickM-27** [reply]
are you setting timezone in the config? I wonder why no one else experiences this issue

**Reply by thinkloop** [reply]
Oh damn, you got it:

```
  # Optional: Set a timezone to use in the UI (default: use browser local time)
  timezone: None
```

Is this from the docs?

I also have this in docker-compose:

```
- /etc/localtime:/etc/localtime:ro
```

**Reply by thinkloop** [reply]
I removed that line from config and all my times are back, ty.

**Reply by NickM-27** [reply]
ah yes, because that object is a string type, that shouldn't be set as it defaults to null

**Reply by thinkloop** [reply]
Should the docs be updated?

![image](https://user-images.githubusercontent.com/118814/228318581-df22f013-0096-487c-a957-3937f1c50f7d.png)

**Reply by NickM-27** [reply]
I don't think so? None is the default value (but when set programmatically, None means null) and as it says at the top, the reference config is not supposed to be copied from directly

**Reply by blakeblackshear** [reply]
I think in other instances we have left settings in the reference config commented out if they shouldn't be set.

**Reply by NickM-27** [reply]
ah gotcha, makes sense. I'll update that along with some other docs things other issues are showing need to be tweaked

**Reply by thinkloop** [reply]
> I don't think so? None is the default value (but when set programmatically, None means null) and as it says at the top, the reference config is not supposed to be copied from directly

Out of curiosity, which runtime makes "None" `null`?

**Reply by NickM-27** [reply]
> > I don't think so? None is the default value (but when set programmatically, None means null) and as it says at the top, the reference config is not supposed to be copied from directly
> 
> 
> 
> Out of curiosity, which runtime makes "None" `null`?

Python

https://realpython.com/null-in-python/

https://github.com/blakeblackshear/frigate/blob/4c5cd02ab7d2b3d55ab8b3ef2496519f65653f73/frigate/config.py#L86

**Reply by banthungprong** [reply]
I had this problem today. 
If I use 
`timezone: Asia/Bangkok` 
(cams are in Thailand) the "Invalid time" disappears and data/time appear with the correct format defined in strftime_fmt.

If I do not use the timezone key/value pair the Invalid time reappears.

Just tried this 3 times on/off. Maybe you can try with a timezone in your config.

**Reply by lucasnz** [reply]
I've created a pull request (https://github.com/blakeblackshear/frigate/pull/6205) that should resolve the "invalid time" error when strftime_fmt is specified but no timezone if set in the config.
